### ISSUE TITLE
best fit with aspect ratio

### ISSUE THREAD

**Issue by MastroLindus** [DominicTobias/react-image-crop#196]
Hi,

first of all thanks for the library, works like a charm and it's really nice!

I have a question.
If I have an image and I have an aspect ratio, I would like to show the crop having that aspect ratio, and that at the same time crops the image in the optimal way, that is:
1)it grabs the biggest region using the aspect ratio (usually meaning that one of the dimension of the crop is the same as the dimension of the image, and the other dimension is calculated to match the ratio)
2)it is centered on the image

Of course given the image size and the aspect ratio, I can calculate the crop for this by myself, I was just wondering if there's already some functionality for it behind some prop as that would be convenient and I would imagine quite a standard way to show the initial crop.

Thank you again for any answer and have a nice day!

**Comment by DominicTobias** [comment]
This doesn't sound hard but also something that doesn't need to be in the library. If you want to know if the image is portrait or landscape then listen for the onImageLoaded event. Depending on that set the width or height to 100 and then the other a ratio of that. Depending on that that variable side is set 'x' to center it.

**Comment by MastroLindus** [comment]
Hi Dominic,

thank you for your answer.
It is not hard at all, I already implemented it in my own user-land code, and it works fine (and indeed I used the onImageLoaded as you also mentioned).
I just thought that an optional bestFit prop to initialize the crop size in case only the aspect property was set would have been quite an elegant solution to something that likely many people will want, as I would argue it is quite a common use case.

Feel free to take it as a feature suggestion or just ignore it as you like, I'll close the issue meanwhile
